packaging/suse/datadog-agent.init

```diff
@@ -0,0 +1,481 @@
+                                     grep -c '\[program:')
+
+        # Number of expected optional supervisord programs (dogstatsd, jmxfetch and go-metro)
+        supervisor_config_opt_programs=$(grep -E $DD_OPT_PROC_REGEX $SUPERVISOR_FILE |
+                                     grep -c '\[program:')
+
+        if [ "$datadog_supervisor_processes" -ne "$supervisor_config_programs" ]; then
+            ESSENTIAL=1
+        fi
+
+        if [ "$datadog_supervisor_opt_processes" -ne "$supervisor_config_opt_programs" ]  ; then
+            OPTIONAL=1
+        fi
+
+        if [ "$QUERY" = "essential" ] && [ "$ESSENTIAL" -eq 0 ]; then
+            FAIL=0
+        elif [ "$QUERY" = "optional" ] && [ "$OPTIONAL" -eq 0 ]; then
+            FAIL=0
+        elif [ "$QUERY" = "all" ] && [ "$ESSENTIAL" = "$OPTIONAL" ] && [ $ESSENTIAL -eq 0 ]; then
+            FAIL=0
+        fi
+
+        if [ "$FAIL" -gt 0 ]; then
+            echo "$supervisor_processes"
+            echo "$DESC (supervisor) is NOT running all child processes"
+            return 1
+        else
+            echo "$DESC (supervisor) is running all child processes"
+            return 0
+        fi
+
+
+    else
+        echo "$DESC (supervisor) is not running"
+        return 1
+    fi
+}
+
+# Action to take
+case "$1" in
+    start)
+        if [ ! -f $AGENTCONF ]; then
+            echo "$AGENTCONF not found. Exiting."
+            exit 3
+        fi
+
+        check_status essential > /dev/null
+        if [ $? -eq 0 ]; then
+            echo "$DESC is already running"
+            exit 0
+        else
+            $0 stop > /dev/null 2>&1
+        fi
+
+        su $AGENTUSER -c "$AGENTPATH configcheck" > /dev/null 2>&1
+        if [ $? -ne 0 ]; then
+            log_daemon_msg "Invalid check configuration. Please run sudo /etc/init.d/datadog-agent configtest for more details."
+            log_daemon_msg "Resuming starting process."
+        fi
+
+
+        log_daemon_msg "Starting $DESC (using supervisord)" "$NAME"
+        PATH=$SYSTEM_PATH /sbin/start-stop-daemon --start --quiet --oknodo --exec $SUPERVISORD_PATH -- -c $SUPERVISOR_FILE --pidfile $SUPERVISOR_PIDFILE
+        if [ $? -ne 0 ]; then
+            log_end_msg 1
+        fi
+
+        # check if the agent is running once per second for 10 seconds
+        retries=10
+        while [ $retries -gt 1 ]; do
+          if check_status essential > /dev/null; then
+              # We've started up successfully. Exit cleanly
+              log_end_msg 0
+              exit 0
+          else
+              retries=$(($retries - 1))
+              sleep 1
+          fi
+        done
+        # After 10 tries the agent didn't start. Report an error
+        log_end_msg 1
+        check_status # report what went wrong
+        $0 stop
+        exit 1
+        ;;
+    stop)
+        log_daemon_msg "Stopping $DESC (stopping supervisord)" "$NAME"
+        /sbin/start-stop-daemon --stop --quiet --oknodo --pidfile $SUPERVISOR_PIDFILE
```

Comment:
I noticed the `debian` counterpart had a `--retry 30` argument for `start-stop-daemon` is that not available in suse?